projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts

```diff
@@ -163,8 +163,14 @@ export class IgxGridGroupByRowComponent {
      * this.grid1.rowList.first.toggle()
      * ```
      */
-    public toggle() {
-        this.grid.toggleGroup(this.groupRow);
+    public toggle(key?) {
+        const shouldExpand = (!key && !this.expanded) || (key && !this.expanded && (key === 'arrowleft' ||  key === 'left'));
+        if (shouldExpand) {
+            this.handleToggleScroll();
```

Comment:
We can probably call this.handleToggleScroll(); before the conditional statement and then just dispatch if !shouldExpand.